test/shopify-cli/changelog_test.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+require "test_helper"
+require "shopify_cli/changelog"
+
+module ShopifyCLI
+  class ChangelogTest < MiniTest::Test
+    TEST_CHANGELOG = <<~CHANGELOG
+      Explanatory note
+
+      ## [Unreleased]
+
+      ### Fixed
+      * [#1234](https://github.com/Shopify/shopify-cli/pull/1234): Fixed 1234
+
+      ## Version #{ShopifyCLI::VERSION}
+
+      ### Added
+      * [#5678](https://github.com/Shopify/shopify-cli/pull/5678): Added 5678
+    CHANGELOG
+
+    def test_changelog_outputs_equal_inputs
+      assert_equal(Changelog.new.full_contents, File.read(Changelog::CHANGELOG_FILE))
+    end
+
+    def test_adding_to_existing_section
```

Comment:
These tests are intended to add confidence that the changelog is being built correctly, i.e.:

1. Changes are added to the `Unreleased` section
2. Changes are added to the correct section, and a new section is opened if necessary.